core/autoscaling.go

```diff
@@ -182,6 +239,8 @@ func (a *autoScalingGroup) process() {
 	a.scanInstances()
 	a.loadDefaultConfig()
 	a.loadConfigFromTags()
+	a.loadDefaultSpotConfig()
+	a.loadSpotConfigFromTags()
```

Comment:
I think those 2 methods could be simplified (by removing the boolean) and put directly in the previous 2 methods (`loadDefaultConfig` & `loadConfigFromTags`) that were created for that; in case extra elements were to be added.

While re-reading the code I also realised it could be slightly re-arranged in order to load the default configuration and ASG configuration earlier than during the processing, see https://github.com/cristim/autospotting/issues/180.

Reply:
loadDefaultConfig has dedicated tests and merging mine with that will be a bit messy. What do you think @cristim ?

Reply:
I  agree with @xlr-8, the purpose of tests is for making such refactoring easier, let's merge these because they do more or less the same thing.